api/src/db/DynamoUserDataClient.ts

```diff
@@ -24,23 +25,32 @@ const unmarshallOptions = {
 
 export const dynamoDbTranslateConfig = { marshallOptions, unmarshallOptions };
 
-const migrateUserData = (data: any): any => {
+const migrateUserData = (data: any, logger: LogWriterType): any => {
+  const logId = logger.GetId();
   const dataVersion = data.version ?? CURRENT_VERSION;
   const migrationsToRun = Migrations.slice(dataVersion);
   const migratedData = migrationsToRun.reduce((prevData: any, migration: MigrationFunction) => {
-    return migration(prevData);
+    try {
+      return migration(prevData);
+    } catch (error) {
+      logger.LogError(`Dynamo User Migration Error - Id:${logId} - Error: ${error}`);
```

Comment:
It might be nice to have some additional helper info in here.
- what user data was the issue (user UUID?)
- what migration # was running when the failure occurred
- what version of user data the user was created with